client/src/pages/components/mailInForm.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import * as dc from './dumbComponents';
+
+class MailInForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            recipients: undefined,
+            recipientsNotSubmitted: true,
+            email: undefined,
+            emailNotSubmitted: true,
+            subject: undefined,
+            subjectNotSubmitted: true,
+            name: undefined,
+            nameNotSubmitted: true,
+            message: undefined,
+            autoGenerateMessage: true,
+            messageNotSubmitted: true,
```

Comment:
I don't think state should be managed with booleans to check if a field has been submitted. Can you just check the value of the field in the state?

if (this.state.subject  === undefined)

Reply:
Agreed